### PR TITLE
[5.0] Credscan 

### PR DESCRIPTION
**Pull request by aik-jahoda**
This is port of https://github.com/dotnet/corefx/pull/43051

for 6.0 - #50230

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]
Please check with @Anipik / @wtgodbe that we can merge asap.

**Comment by wtgodbe** [comment]
Merging ASAP is ok, but again please make sure to cancel the triggered internal builds of `release/5.0` _and_ `internal/release/5.0`

File: src/libraries/System.Security.Cryptography.Xml/tests/EncryptedXmlTest.cs [hunk 1]
```diff
@@ -139,10 +139,10 @@ public void Sample2()
         {
             using (Aes aes = Aes.Create())
             {
+                byte[] keydata = convert.FromBase64String("o/ilseZu+keLBBWGGPlUHweqxIPc4gzZEFWr2nBt640=");
```
Comment (danmoseley) [inline src/libraries/System.Security.Cryptography.Xml/tests/EncryptedXmlTest.cs:142]:
```suggestion
                byte[] keydata = Convert.FromBase64String("o/ilseZu+keLBBWGGPlUHweqxIPc4gzZEFWr2nBt640=");
```

File: src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs [hunk 2]
```diff
@@ -520,46 +520,46 @@ private static IEnumerable<KnownValuesTestCase> GetKnownValuesTestCases()
                 CaseName = "SHA256 alternate",
                 HashAlgorithmName = "SHA256",
                 // [SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="Unit test dummy credentials.")]
```
Comment (danmoseley) [inline src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs:522]: Remove suppressions like this?

**Review by danmoseley — APPROVED** [review]

File: src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs [hunk 3]
```diff
@@ -520,46 +520,46 @@ private static IEnumerable<KnownValuesTestCase> GetKnownValuesTestCases()
...
                     // T-Block 2
-                    "27B792394D6C13DB121CD16683CD738CB1717C69B34EF2B29E32306D24FCDF"),
+                    "EE6DB97E5852FC4C15FA7C52FACDEDE89B916BCC864028084A2CF0889F7F76"),
             };
 
             yield return new KnownValuesTestCase
             {
                 CaseName = "SHA384 alternate",
                 HashAlgorithmName = "SHA384",
                 // [SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="Unit test dummy credentials.")]
```
Comment (danmoseley) [inline src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs:537]: Remove unnecessary suppression

File: src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs [hunk 4]
```diff
@@ -520,46 +520,46 @@ private static IEnumerable<KnownValuesTestCase> GetKnownValuesTestCases()
...
                     // T-Block 2
-                    "AC02F63AA2B0EC9697B1801E70BD10A6B58CE5DE83DD18F4FFD2E8D9289716510AA0A170EF1D145F4B3247"),
+                    "3C6A08364D0A47CD1F15E0E314800FF3AC9CF2E93B3F81A5EB67FE9F2FE6E86B0430B59902CCB5FD190E67"),
             };
 
             yield return new KnownValuesTestCase
             {
                 CaseName = "SHA512 alternate",
                 HashAlgorithmName = "SHA512",
                 // [SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="Unit test dummy credentials.")]
```
Comment (danmoseley) [inline src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs:552]: Remove unnecessary suppression

**Comment by danmoseley** [comment]
`System.Security.Cryptography.X509Certificates.Tests` failure is the occasional sporadic OS error we get on Windows.

**Comment by danmoseley** [comment]
BSD run seems jammed. I'm not going to wait on that as the probability that this introduces a BSD-unique issue is vanishingly small.

**Pull request merged**